### PR TITLE
Added wallet crypto trading accounts

### PR DESCRIPTION
**Pull request by zubair-deriv**
https://app.clickup.com/t/20696747/WALL-4245

### CODE REVIEW

File: cypress/e2e/smoke/Wallets/addTradingAccount.cy.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+  )
+
+  it(
+    'should add trading account to wallet account (Crypto) in responsive',
+    { scrollBehavior: false },
+    () => {
+      cy.c_visitResponsive('/', 'small')
+      cy.c_skipPasskeysV2()
+      cy.contains('Wallet', { timeout: 10000 }).should('exist')
```
Comment (bilalmajid-deriv) [inline cypress/e2e/smoke/Wallets/addTradingAccount.cy.js:26]: Can we not use a testing library locator here

File: cypress/e2e/smoke/Wallets/addTradingAccount.cy.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  beforeEach(() => {
+    cy.c_login({ user: 'walletloginEmail' })
+  })
+
+  it(
+    'should add trading account to wallet account (Crypto)',
+    { scrollBehavior: false },
+    () => {
+      cy.c_visitResponsive('/', 'large')
+      cy.contains('Wallet', { timeout: 10000 }).should('exist')
```
[the remark this answers is not in the document]
Comment (zubair-deriv) [reply cypress/e2e/smoke/Wallets/addTradingAccount.cy.js:13]: yes, updated to findByText

File: cypress/support/commands/wallets.js [hunk 3]
```diff
@@ -86,3 +86,41 @@ Cypress.Commands.add('c_checkForBanner', () => {
...
+          .and('be.enabled')
+          .click()
+        cy.contains("Trader's Hub").should('be.visible')
+      }
+    })
+})
+
+Cypress.Commands.add('c_setupTradeAccountResponsive', (wallet) => {
+  cy.findByRole('button', { name: 'Options' }).click()
+  // cy.wait(2000)
```
[the remark this answers is not in the document]
Comment (zubair-deriv) [reply cypress/support/commands/wallets.js:110]: updated

File: cypress/support/commands/wallets.js [hunk 4]
```diff
@@ -86,3 +86,41 @@ Cypress.Commands.add('c_checkForBanner', () => {
...
+  cy.findByRole('button', { name: 'Get' })
+    .should(() => {})
+    .then((button) => {
+      if (button.length) {
+        cy.wrap(button).click()
+        // cy.wait(1000)
```
[the remark this answers is not in the document]
Comment (zubair-deriv) [reply cypress/support/commands/wallets.js:116]: updated

File: cypress/e2e/smoke/Wallets/addTradingAccount.cy.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+  it(
+    'should add trading account to wallet account (Crypto) in responsive',
+    { scrollBehavior: false },
+    () => {
+      cy.c_visitResponsive('/', 'small')
+      cy.c_skipPasskeysV2()
+      cy.contains('Wallet', { timeout: 10000 }).should('exist')
+      cy.c_skipPasskeysV2()
+      cy.contains('Deposit', { timeout: 10000 }).should('exist')
```
[the remark this answers is not in the document]
Comment (zubair-deriv) [reply cypress/e2e/smoke/Wallets/addTradingAccount.cy.js:28]: yes, updated to findByText

**Review by bilalmajid-deriv — APPROVED** [review]
lgtm

File: cypress/e2e/smoke/Wallets/cryptoTransactions.cy.js [hunk 6]
```diff
@@ -31,23 +31,6 @@ function crypto_transfer(to_account, transferAmount) {
   cy.findByRole('button', { name: 'Make a new transfer' }).click()
 }
 
-function setupTradeAccount(wallet) {
```
Comment (maliheh-deriv) [inline cypress/e2e/smoke/Wallets/cryptoTransactions.cy.js:34]: please make sure that this test `cryptoTransactions` is updated with cy.c_setupTradeAccount as well.

File: cypress/support/commands/wallets.js [hunk 7]
```diff
@@ -86,3 +86,39 @@ Cypress.Commands.add('c_checkForBanner', () => {
...
+    .then((button) => {
+      if (button.length) {
+        cy.wrap(button).click()
+        cy.wait(1000)
+        cy.findByRole('button', { name: 'Transfer funds' }).should('be.visible')
+        cy.findByRole('button', { name: 'Maybe later', timeout: 3000 })
+          .should('be.visible')
+          .and('be.enabled')
+          .click()
+        cy.contains("Trader's Hub").should('be.visible')
```
Comment (maliheh-deriv) [inline cypress/support/commands/wallets.js:103]: why we have not used cy.findByText("Trader's Hub") here ?

Comment (zubair-deriv) [reply cypress/support/commands/wallets.js]: cy.findByText wasn't working as we have similar text on the page. I have updated the validation using  cy.findByTestId

File: cypress/e2e/smoke/Wallets/cryptoTransactions.cy.js (hunk 6 above)
Comment (zubair-deriv) [reply cypress/e2e/smoke/Wallets/cryptoTransactions.cy.js:34]: updated

File: cypress/e2e/smoke/Wallets/addTradingAccount.cy.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import '@testing-library/cypress/add-commands'
+
+describe('QATEST-156146 - Add trading account to wallet account (Crypto)', () => {
+  beforeEach(() => {
+    cy.c_login({ user: 'walletloginEmail' })
```
[the remark this answers is not in the document]
Comment (zubair-deriv) [reply cypress/e2e/smoke/Wallets/addTradingAccount.cy.js:5]: yes, tested the behavior, and this code has no impact on existing specs.

**Review by maliheh-deriv — APPROVED** [review]

**Pull request merged**